sdk/search/azure-search/src/main/java/com/azure/search/documents/implementation/util/CustomPatternTokenizerDeserializer.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.util;
+
+import com.azure.search.documents.models.PatternTokenizer;
+import com.azure.search.documents.models.RegexFlags;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Custom deserializer for {@link PatternTokenizer}, which convert flatten string to the list of {@link RegexFlags}.
+ */
+public class CustomPatternTokenizerDeserializer extends JsonDeserializer<PatternTokenizer> {
+    private static final String DELIMITER = "\\|";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PatternTokenizer deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode root = mapper.readTree(p);
+        JsonNode flattenFlags = root.get("flags");
+        List<RegexFlags> regexFlags = Arrays.stream(flattenFlags.asText().split(DELIMITER))
+            .map(RegexFlags::fromString).collect(Collectors.toList());
+        Iterator<Map.Entry<String, JsonNode>> fields = root.fields();
+        PatternTokenizer tokenizer = new PatternTokenizer();
+        while (fields.hasNext()) {
+            Map.Entry<String, JsonNode> field = fields.next();
+            if ("name".equals(field.getKey())) {
+                tokenizer.setName(field.getValue().asText());
+            } else if ("pattern".equals(field.getKey())) {
+                tokenizer.setPattern(field.getValue().asText());
+            } else if ("flags".equals(field.getKey())) {
+                tokenizer.setFlags(regexFlags);
+            } else if ("group".equals(field.getKey())){
+                tokenizer.setGroup(field.getValue().asInt());
```

Comment:
How does `asInt()` handle if `getValue()` is `null`? `group` is a nullable `Integer` type so it could come back as `null` from my understanding.

Reply:
I don't think the case exists.
If the field value is null, then there is no need to generate field key. If the case exists for required field, then we need to have null or empty checking for the field.